pkg/deploy/controller/deployerpod/controller.go

```diff
@@ -97,14 +97,8 @@ func (c *DeployerPodController) Handle(pod *kapi.Pod) error {
 			nextStatus = deployapi.DeploymentStatusRunning
 		}
 	case kapi.PodSucceeded:
-		// Detect failure based on the container state
 		nextStatus = deployapi.DeploymentStatusComplete
-		for _, info := range pod.Status.ContainerStatuses {
```

Comment:
@ironcladlou @smarterclayton I am removing this as [kubelet](https://github.com/kubernetes/kubernetes/blob/f89432e44b7aeac6e48d16fc34952e644a388b8d/pkg/kubelet/kubelet.go#L3230) already does that for us.